storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1382,5 +1422,438 @@ public static TopologyInfo getTopologyInfo(String name, String asUser, Map storm
+//                newMap.put(key, value);
+//            }
+//        }
+//        return newMap;
+//    }
+
+    /* TODO: This function was originally written to replace map-key in util.clj. But we decided to change the coding
+         style in the caller functions to a more Java style for loop. This is mentioned in TODOs across the clojure
+         files.
+    */
+//    public static Map mapKey (IFn aFn, Map amap) {
+//        Map newMap = new HashMap();
+//        if (amap == null) {
+//            return newMap;
+//        }
+//        for (Object key: amap.keySet()) {
+//            Object value = amap.get(key);
+//            Object newKey = aFn.eval(key);
+//            newMap.put(newKey, value);
+//        }
+//        return newMap;
+//    }
+
+    /* TODO: This function was originally written to replace separate in util.clj. But since it was only used in
+    transactional_test.clj, the separate function was moved there for now.
+    */
+    public static Vector<Collection> separate (IPredicate pred, Collection aseq) {
+        Vector<Collection> outputVector = new Vector<Collection>();
+        Collection pass = new HashSet();
+        Collection notPass = new HashSet();
+        for (Object obj: aseq) {
+            if (pred.test(obj)) {
+                pass.add(obj);
+            } else {
+                notPass.add(obj);
+            }
+        }
+        outputVector.add(pass);
+        outputVector.add(notPass);
+        return outputVector;
+    }
+
+    public static void exitProcess (int val, Object... msg) {
+        String combinedErrorMessage = "";
+        for (Object oneMessage: msg) {
+            combinedErrorMessage = combinedErrorMessage + oneMessage.toString();
+        }
+        LOG.error("halting process: " + combinedErrorMessage, new RuntimeException(combinedErrorMessage));
+        Runtime.getRuntime().exit(val);
+    }
+
+    public static Double sum(Collection<Number> vals) {
+        double sum = 0.0;
+        double dVal;
+        if (vals == null) {
+            return 0.0;
+        }
+        for (Number val: vals) {
+            dVal = val.doubleValue();
+            sum = sum + dVal;
+        }
+        return new Double(sum);
+    }
+
+
+    public static Object defaulted(Object val, Object defaultObj) {
+        if (val != null) {
+            return val;
+        } else {
+            return defaultObj;
+        }
+    }
+
+    /**
+     * Deletes a file or directory and its contents if it exists. Does not
+     * complain if the input is null or does not exist.
+     * @param path the path to the file or directory
+     */
+    public static void forceDelete(String path) {
+        _instance.forceDeleteImpl(path);
+    }
+
+    protected void forceDeleteImpl(String path) {
+        LOG.debug("Deleting path {}", path);
+        if (checkFileExists(path)) {
+            try {
+                FileUtils.forceDelete(new File(path));
+            } catch (IOException ignored) {}
```

Comment:
FileNotFoundException not IOException.  Other IOExceptions are thrown, like I don't have permission to delete this.  FileNotFound is I lost a race and don't need to delete it.